operator/pkg/tpath/tpath.go

```diff
@@ -141,6 +141,27 @@ func getPathContext(nc *PathContext, fullPath, remainPath util.Path, createMissi
 				}
 				continue
 			}
+			if lm, ok := le.(map[string]interface{}); ok {
```

Comment:
no clue what this does, so hard for me to review

Reply:
this is an unfortunate side effect of yaml unmarshal and type assertions. yaml unmarshal produces trees that mix map[string]interface{} and map[interface{}]interface{} and you need to handle both types to traverse the tree. unfortunately there's no clean way to combine the code blocks to avoid code duplication that i know of because the type asserted symbol inside the block can only have one type that is decided at compile time. the only way around that is to use reflect instead of type assertions but since we've gone down the type assertion path elsewhere it would be inconsistent to use reflect only here.